### ISSUE TITLE
Rule S09b test data discrepancy

### ISSUE THREAD

**Issue by mhagemann-eab** [ut-effectiveness/utValidateR#8]
``` r
library(utValidateR)
testdf <- get_test_data(file = "student")
#> Warning in check_expected_values(., colname = expected_value_column): The following rows of test data have bad expected values and were removed: 62, 63, 64, 65, 66
#> Warning in check_rule_names(., checklist = checklist, colname = rule_name_column): The following rows of test data have bad rule names and were removed: 25, 42, 43
knitr::kable(compare_rule_output("S09b", testdf = testdf))
```

| csv_row | rule | description                                           | expr                                                   | is_international | us_citizenship_code | expected | actual |
|--------:|:-----|:------------------------------------------------------|:-------------------------------------------------------|:-----------------|:--------------------|:---------|:-------|
|       2 | S09b | missing citizenship code (Nonresident Alien mismatch) | equivalent(is_international, us_citizenship_code == 2) | TRUE             | 3                   | pass     | fail   |
|       3 | S09b | missing citizenship code (Nonresident Alien mismatch) | equivalent(is_international, us_citizenship_code == 2) | TRUE             | 9                   | pass     | fail   |
|       4 | S09b | missing citizenship code (Nonresident Alien mismatch) | equivalent(is_international, us_citizenship_code == 2) | FALSE            | NA                  | pass     | fail   |
|       9 | S09b | missing citizenship code (Nonresident Alien mismatch) | equivalent(is_international, us_citizenship_code == 2) | TRUE             | 5                   | pass     | fail   |
|      10 | S09b | missing citizenship code (Nonresident Alien mismatch) | equivalent(is_international, us_citizenship_code == 2) | TRUE             | NA                  | pass     | fail   |
|      11 | S09b | missing citizenship code (Nonresident Alien mismatch) | equivalent(is_international, us_citizenship_code == 2) | TRUE             | 6                   | pass     | fail   |
|      12 | S09b | missing citizenship code (Nonresident Alien mismatch) | equivalent(is_international, us_citizenship_code == 2) | FALSE            | 2                   | pass     | fail   |
|      14 | S09b | missing citizenship code (Nonresident Alien mismatch) | equivalent(is_international, us_citizenship_code == 2) | FALSE            | NA                  | pass     | fail   |
|      16 | S09b | missing citizenship code (Nonresident Alien mismatch) | equivalent(is_international, us_citizenship_code == 2) | FALSE            | NA                  | pass     | fail   |
|      18 | S09b | missing citizenship code (Nonresident Alien mismatch) | equivalent(is_international, us_citizenship_code == 2) | FALSE            | 2                   | pass     | fail   |

<sup>Created on 2022-09-20 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by mhagemann-eab** [comment]
Based on the (complicated!) sql code, I interpret this rule to be verifying that `citizenship_code` should be 2 if and only if `is_international` is TRUE, and if one is missing then the other should be missing as well. But I might be misinterpreting.

**Comment by andreabr-dixie** [comment]
These are more complex.  I will resolve these next week.

**Comment by andreabringhurst** [comment]
I don't think this has been resolved. I was creating a file for bad data so we could test if the rule was working. We will need to determine if I need to go in and work on this file.

**Comment by andreabringhurst** [comment]
The way this rule is written doesn't work because is_international = TRUE when us_citizenship_code = 2 and visa_type = 'F-1'. So, this rule needs to be changed to use visa_type instead of is_international.
